src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -77,24 +86,80 @@ static DamageTable D_8098CDB8 = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-#endif
-
-extern ColliderSphereInit D_8098CD80;
-extern CollisionCheckInfoInit2 D_8098CDAC;
-extern DamageTable D_8098CDB8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CA20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CAD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CBDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Draw.s")
+void func_8098CA20(EnSi* this, PlayState* play) {
+    this->colliderSphere.dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->colliderSphere.dim.worldSphere.center.y = this->actor.world.pos.y;
+    this->colliderSphere.dim.worldSphere.center.z = this->actor.world.pos.z;
+    this->colliderSphere.dim.worldSphere.radius =
+        this->colliderSphere.dim.modelSphere.radius * this->colliderSphere.dim.scale;
+    if (this->actor.colChkInfo.health > 0) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderSphere.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
+}
+
+void func_8098CAD0(EnSi* this, PlayState* play) {
+    s32 chestFlag = ENSI_GET_CHEST_FLAG(&this->actor);
+
+    if ((chestFlag < 0x20) && (chestFlag >= 0)) {
+        Flags_SetTreasure(play, chestFlag);
+    }
+    Item_Give(play, ITEM_SKULL_TOKEN);
+    if (Inventory_GetSkullTokenCount(play->sceneId) >= 30) {
+        Message_StartTextbox(play, 0xFC, NULL);
+        Audio_PlayFanfare(NA_BGM_GET_ITEM | 0x900);
+    } else {
+        Message_StartTextbox(play, 0x52, NULL);
+        Audio_PlayFanfare(NA_BGM_GET_SMALL_ITEM);
+    }
+}
+
+void func_8098CB70(EnSi* this, PlayState* play) {
+    if (CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_2000)) {
+        this->actionFunc = func_8098CBDC;
+    } else if (this->colliderSphere.base.ocFlags2 & OC2_HIT_PLAYER) {
+        func_8098CAD0(this, play);
+        Actor_Kill(&this->actor);
+        return;
+    }
+    this->actor.shape.rot.y += 0x38E;
+}
+
+void func_8098CBDC(EnSi* this, PlayState* play) {
+    if (!CHECK_FLAG_ALL(this->actor.flags, ACTOR_FLAG_2000)) {
```

Comment:
This flag is for being held by the hookshot, so probably `EnSi_DraggedByHookshot` or something like that?